pkg-manager/plugin-commands-installation/test/update/recursive.ts

```diff
@@ -415,3 +416,33 @@ test('recursive update in workspace should not add new dependencies', async () =
   projects['project-1'].hasNot('is-positive')
   projects['project-2'].hasNot('is-positive')
 })
+
+test('recursive update with aliased workspace dependency (#7975)', async () => {
+  const projects = preparePackages([
+    {
+      name: 'project-1',
+      version: '1.0.0',
+      dependencies: {
+        pkg: 'workspace:project-2@^',
+      },
+    },
+    {
+      name: 'project-2',
+      version: '1.0.0',
+    },
+  ])
+
+  await update.handler({
+    ...DEFAULT_OPTS,
+    ...await readProjects(process.cwd(), []),
+    depth: 0,
+    dir: process.cwd(),
+    recursive: true,
+    workspaceDir: process.cwd(),
+  })
+
+  projects['project-1'].has('pkg')
+
+  const manifest = await readProjectManifestOnly('project-1')
+  expect(manifest).toHaveProperty(['dependencies', 'pkg'], 'workspace:npm:project-2@^') // TODO: how to remove the `npm:` part?
```

Comment:
this should be fixed

Reply:
This bug never surface before because the main bug prevents writing to workspace dependency fields. But now it has appear, I don't have direction in how to fix it.